lib/Sema/TypeCheckType.cpp

```diff
@@ -4794,8 +4794,10 @@ NeverNullType TypeResolver::resolveTupleType(TupleTypeRepr *repr,
   if (moveOnlyElementIndex.has_value()
       && !options.contains(TypeResolutionFlags::SILType)
       && !ctx.LangOpts.hasFeature(Feature::MoveOnlyTuples)) {
-    diagnose(repr->getElementType(*moveOnlyElementIndex)->getLoc(),
-             diag::tuple_move_only_not_supported);
+    auto noncopyableTy = elements[*moveOnlyElementIndex].getType();
```

Comment:
It looks like this code path is not tested -- you need a type representation containing a non-copyable type

Reply:
I checked and it turns out to be covered in `Constraints/moveonly_tuple.swift` with this:

```
@_moveOnly
struct Foo {
    var t: (Int, Butt) // expected-error{{tuple with noncopyable element type 'Butt' is not supported}}
}
```